kommitters/calendso
# Update and delete bookings from Google Calendar

## What does this PR do?



Fixes #2009 #2434

I noticed that this issue happens when you change the destination calendar, so if you saved a reservation in the "Primary" calendar and then changed the destination calendar to "Birthday" when you want to reschedule or cancel the booking, the app will try to look for the booking in `Birthday` because that is the target calendar you have at the moment.

Fixed bug when an event is rescheduled or canceled, it is updated only in the database and not in Google Calendar.
To solve this issue I edited the Booking model in `packages/prisma/schema.prisma`, and CalendarEvent 
 interface in `packages/types/Calendar.d.ts`, adding to them `externalIdCalendar`  element, this is going to save the id from the calendar where the booking was saved.



## Type of change




## How should this be tested?



- Create a booking with the primary calendar and go to Google Calendar to check.
- Change the destination calendar.
- Try to cancel or reschedule the booking. Go to Google Calendar and you going to see the booking was updated or deleted
